azure/ConsiderationsForServiceDesign.md

```diff
@@ -77,6 +77,71 @@ Understanding how your service is used and defining its model and interaction pa
 
 :white_check_mark: **DO** create an [OpenAPI Definition](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/2.0.md) (with [autorest extensions](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md)) describing the service. The OpenAPI definition is a key element of the Azure SDK plan and is essential for documentation, usability and discoverability of services.
 
+## Use Good Names
+
+Good names for resources, properties, operations, and parameters are essential to a good developer experience.
+
+Resources are described by nouns. Resource and property names must be descriptive and easy for customers to understand. Ease of understanding comes from familiarity and recognition; you should favor consistency with other Azure services, names in the product's portal/user interface, and industry standards.
+
+Names should aid developers in discovering functionality without having to constantly refer to documentation.
+Use common patterns and standard conventions to aid developers in correctly guessing common property names and meanings.
+Use verbose naming patterns and avoid abbreviations other than
+well-known acronyms in your service domain.
+
```

Comment:
Another common problem I see is that too many concept names are included in APIs. When a service/feature is being designed, the designers need many concept names to be able to debate and reason about detail of the design. Once the feature design is finished, concepts that are not absolutely required to be understood by users should not appear in the finished product, but they often do, yet users are not interested in knowing how the sausage is made.

Reply:
When I see a concept heavy name, I ask the designer "what is blah-foo-bar-baz method doing?". They often describe the semantics using plain English, and when they do I tell them to name the API just what they said.

Reply:
Good points Krzysztof. I have tried to incorporate these but happy to take additional feedback.